File: src/org/elixir_lang/beam/psi/stubs/TypeDefinitionStub.java
package org.elixir_lang.beam.psi.stubs;

import com.intellij.psi.stubs.StubElement;
import org.elixir_lang.beam.psi.TypeDefinition;
import org.jetbrains.annotations.NotNull;

public interface TypeDefinitionStub<T extends TypeDefinition> extends StubElement<T> {
    @NotNull
    String getName();
}

File: src/org/elixir_lang/beam/psi/stubs/TypeDefinitionStub.kt
package org.elixir_lang.beam.psi.stubs

import com.intellij.psi.stubs.StubElement
import org.elixir_lang.beam.psi.TypeDefinition
import org.elixir_lang.type.Visibility

interface TypeDefinitionStub<T : TypeDefinition?> : StubElement<T> {
    val visibility: Visibility
    val name: String
    val arity: Int
}
